### PR TITLE
Make F60addforemanrepo a static hook & include GPG

### PR DESCRIPTION
**Pull request by ekohl**
This rewrites to hook to a pure bash file without any ERB templating. The $DISTRIBUTION variable is guaranteed to be available by pbuilder. It also uses heredoc for a more readable script. The hook already sets up the Foreman repositories, having the GPG setup in the same place keeps it local.

### CODE REVIEW

File: puppet/modules/slave/files/pbuilder_F60addforemanrepo [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	DOGPHe045uwpChLRyslY6qYm0WlhBgWIAFr8dzhAyOhuWz3I6Iw43DACLa+co+WR
+	FZ33FqYTbl+F5x+x983alzXG0+8/eJ0fdcEOPj+C/GbJaKwXjpQk3l7OpUwLIQNy
+	nFVRXYaD1g2i6pHgWhhqFlWqo6fHrbD2Hzm9HCNS5eWiB/JGFZKyGBpBMh22/k6O
+	bIZ8SVgErE+AwvAE7HCnr+Pun/Iblrlb6fLC6J4WHumvmAIPLznIViUekC0jvEym
+	kNo8dY3LUn8NlQJ/qtBZDNf/OvP5LjiHPJdLlTyc+eZDGIszkPwLO/rjUfJGaD20
+	1vYLOIp+VREV4ODtMQfkLv7W+ynhzAgOZJBiJWU0CwLbnWza4D+yIIc=
+	=Fyc2
+	-----END PGP PUBLIC KEY BLOCK-----
+	-----BEGIN PGP PUBLIC KEY BLOCK-----
+	mQINBGD6Y/UBEADOhGOD74lvHdUAew1cHjAUTbopfoXvFvo5jXs4ra3IblnzPEOJ
```
Comment (evgeni) [inline puppet/modules/slave/files/pbuilder_F60addforemanrepo:67]: you lost a newline between the "begin block" and the actual content, and I know from painful experience, that `apt` doesn't like that (gpg does)

**Comment by evgeni** [comment]
```
jenkins@deb-node01:~$ FOREMAN_VERSION=nightly sudo pbuilder-buster64 login
…
root@deb-node01:/# echo $DISTRIBUTION
<empty>
```

**Comment by ekohl** [comment]
From `man pbuilder`:
```
       --hookdir [location of user scripts]

              [...]

              In your hook the following environment variables are available:
               PBUILDER_OPERATION a string indicating which of the pbuilder command has been called.  Possible values are build, clean, create, debuild, dumpconfig, execute, lo‐
               gin, update.
               BUILDDIR the place where the build happens, the sources are inside a directory named <package>-<version>.
               DISTRIBUTION the name of the used distribution, as provided by the DISTRIBUTION config value, or the --distribution command line flag.
               BUILD_ARCH contains the build architecture, the architecture the package is building on.
               HOST_ARCH contains the host architecture, the architecture the package is building for.
```

I trusted this.

**Comment by evgeni** [comment]
Hmm, it's only available in the hooks? maybe.

**Comment by evgeni** [comment]
Tried, available in hooks, thanks

File: puppet/modules/slave/files/pbuilder_F60addforemanrepo [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+if [[ -n ${FOREMAN_VERSION:-} ]]; then
+	cat > /etc/apt/sources.list.d/foreman.list <<-EOF
+	deb http://deb.theforeman.org/ ${DISTRIBUTION} ${FOREMAN_VERSION}
+	deb http://deb.theforeman.org/ plugins ${FOREMAN_VERSION}
+	deb http://stagingdeb.theforeman.org/ ${DISTRIBUTION} theforeman-${FOREMAN_VERSION}
+	EOF
+
+	# The empty line after BEGIN PGP PUBLIC is needed by apt
+	cat > /etc/apt/trusted.gpg.d/foreman.gpg <<-EOF
```
Comment (evgeni) [inline puppet/modules/slave/files/pbuilder_F60addforemanrepo:14]:
```suggestion
	cat > /etc/apt/trusted.gpg.d/foreman.asc <<-EOF
```

**Review by evgeni — CHANGES_REQUESTED** [review]

File: puppet/modules/slave/files/pbuilder_F60addforemanrepo (hunk 2 above)
Comment (ekohl) [reply puppet/modules/slave/files/pbuilder_F60addforemanrepo:14]: Bleh, rebases gone wrong.

**Comment by ekohl** [comment]
https://github.com/theforeman/foreman-infra/pull/1646 includes this. I'd also be happy if you merge that instead :)

**Review by evgeni — APPROVED** [review]

**Pull request merged**